newrelic/hooks/mlmodel_openai.py

```diff
@@ -1245,6 +1275,25 @@ async def wrap_engine_api_resource_create_async(wrapped, instance, args, kwargs)
         return return_val
 
 
+async def wrap_stream_iter_events_async(wrapped, instance, args, kwargs):
+    transaction = current_transaction()
+    if not transaction:
+        async_iterator = wrapped(*args, **kwargs)
+        async for return_val in async_iterator:
+            yield return_val
+        return
+
+    async_iterator = wrapped(*args, **kwargs)
+    try:
+        async for return_val in async_iterator:
+            record_stream_chunk(instance, return_val)
+            yield return_val
+    except Exception as exc:
+        record_error(instance, transaction, exc)
+        raise
+    record_events_on_stop_iteration(instance, transaction)
```

Comment:
I'm not positive but this might be catching the StopAsyncIteration exception as well. I think you'd need another except clause above this one to handle it.

Reply:
Disregard that, `except Exception` does not include StopAsyncIteration so this is fine.